contracts/Vault.vy

```diff
@@ -764,6 +768,7 @@ def withdraw(_shares: uint256 = MAX_UINT256, _recipient: address = msg.sender) -
         caller's address.
     @return The quantity of tokens redeemable for `_shares`.
     """
+    assert block.number > self.lastDeposit[msg.sender]
```

Comment:
At first glance I like it, is this is easy to test for in ganache fork?

Reply:
We can create a "flash loan" mock and show that it fails

Reply:
Yes, please.

Reply:
Suggested change from today's chat:
```suggestion
    assert block.timestamp > self.lastReport
```

We don't need `lastDeposit`.

The way this will work is that it'll disallow sandwich attacks via flash loans (or normal txn sandwiching) in the same block.